src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderValue.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Formats.Cbor;
+
+namespace System.Security.Cryptography.Cose
+{
+    public readonly struct CoseHeaderValue : IEquatable<CoseHeaderValue>
+    {
+        public readonly ReadOnlyMemory<byte> EncodedValue { get; }
+
+        private CoseHeaderValue(ReadOnlyMemory<byte> encodedValue)
+        {
+            EncodedValue = encodedValue;
+        }
+
+        public static CoseHeaderValue FromEncodedValue(ReadOnlySpan<byte> encodedValue)
+        {
+            // We don't validate here as we need to know in which label the value is going to be used to validate even more semantics.
+            var encodedValueCopy = new ReadOnlyMemory<byte>(encodedValue.ToArray());
+            CoseHeaderValue value = new CoseHeaderValue(encodedValueCopy);
+            return value;
+        }
+
+        public static CoseHeaderValue FromEncodedValue(byte[] encodedValue)
+            => FromEncodedValue(encodedValue.AsSpan());
+
+        public static CoseHeaderValue FromInt32(int value)
+        {
+            var writer = new CborWriter();
+            writer.WriteInt32(value);
+
+            return FromEncodedValue(writer.Encode());
+        }
+
+        public static CoseHeaderValue FromString(string value)
+        {
+            var writer = new CborWriter();
+            writer.WriteTextString(value);
+
+            return FromEncodedValue(writer.Encode());
+        }
+
+        public static CoseHeaderValue FromBytes(ReadOnlySpan<byte> value)
+        {
+            var writer = new CborWriter();
+            writer.WriteByteString(value);
+
+            return FromEncodedValue(writer.Encode());
+        }
+
+        public static CoseHeaderValue FromBytes(byte[] value)
+        {
+            var writer = new CborWriter();
+            writer.WriteByteString(value);
+
+            return FromEncodedValue(writer.Encode());
+        }
+
+        public int GetValueAsInt32()
```

Comment:
Since the API is approved now, and we're moving toward final release, all of the API members should have xmldoc.  At this point it probably makes sense to do the whole library in one go as a followup, but the act of documenting will actually help with things like "what exceptions should we be throwing here?"